### PR TITLE
enhance: clean compaction task in compactionHandler

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #35711

### CODE REVIEW

File: internal/datacoord/compaction.go [hunk 1]
```diff
@@ -709,9 +719,40 @@ func (c *compactionPlanHandler) checkCompaction() error {
...
 	}
 	c.executingGuard.Unlock()
+
+	// insert task need to clean
+	c.cleaningGuard.Lock()
+	for _, t := range finishedTasks {
+		if t.GetTaskProto().GetState() == datapb.CompactionTaskState_failed ||
+			t.GetTaskProto().GetState() == datapb.CompactionTaskState_timeout ||
+			t.GetTaskProto().GetState() == datapb.CompactionTaskState_completed {
+			c.cleaningTasks[t.GetTaskProto().GetPlanID()] = t
```
Comment (czs007) [inline internal/datacoord/compaction.go:729]: add a info log

Comment (czs007) [reply internal/datacoord/compaction.go:729]: in the log print the taskID and collectionID

File: internal/datacoord/compaction_task_clustering.go [hunk 2]
```diff
@@ -151,26 +161,33 @@ func (t *clusteringCompactionTask) retryableProcess() error {
...
-	case datapb.CompactionTaskState_timeout:
-		return t.processFailedOrTimeout()
-	case datapb.CompactionTaskState_failed:
-		return t.processFailedOrTimeout()
+		return t.processStats(ctx)
 	}
 	return nil
 }
 
+func (t *clusteringCompactionTask) Clean() bool {
```
Comment (czs007) [inline internal/datacoord/compaction_task_clustering.go:179]: just return t.doClean() == nil
the log is in t.doClean.

File: internal/datacoord/compaction_task_clustering.go [hunk 3]
```diff
@@ -527,87 +550,107 @@ func (t *clusteringCompactionTask) resetSegmentCompacting() {
...
-		log.Warn("clusteringCompactionTask processFailedOrTimeout unable to drop compaction plan", zap.Int64("planID", t.GetTaskProto().GetPlanID()), zap.Error(err))
+		log.Warn("clusteringCompactionTask unable to drop compaction plan", zap.Error(err))
 	}
-	isInputDropped := false
-	for _, segID := range t.GetTaskProto().GetInputSegments() {
-		if t.meta.GetHealthySegment(context.TODO(), segID) == nil {
-			isInputDropped = true
-			break
+	if t.GetTaskProto().GetState() == datapb.CompactionTaskState_completed {
+		if err := t.markInputSegmentsDropped(); err != nil {
```
Comment (czs007) [inline internal/datacoord/compaction_task_clustering.go:567]: Is this operation idempotence?

File: internal/datacoord/compaction_task_clustering.go [hunk 4]
```diff
@@ -527,87 +550,107 @@ func (t *clusteringCompactionTask) resetSegmentCompacting() {
...
-	log.Info("clean task", zap.Int64("triggerID", t.GetTaskProto().GetTriggerID()), zap.Int64("planID", t.GetTaskProto().GetPlanID()), zap.String("state", t.GetTaskProto().GetState().String()))
+	return nil
+}
+
+func (t *clusteringCompactionTask) doClean(ctx context.Context) error {
+	log := log.Ctx(ctx).With(zap.Int64("planID", t.GetTaskProto().GetPlanID()))
+	log.Info("clean task", zap.Int64("triggerID", t.GetTaskProto().GetTriggerID()),
+		zap.String("state", t.GetTaskProto().GetState().String()))
 
 	if err := t.sessions.DropCompactionPlan(t.GetTaskProto().GetNodeID(), &datapb.DropCompactionPlanRequest{
```
Comment (czs007) [inline internal/datacoord/compaction_task_clustering.go:561]: It is imperative to ensure that the DataNode receiver processes this Drop operation idempotently. Additionally, it is essential to capture the merr.ErrNodeNotFound error and consider the DropCompactionPlan operation complete, as the Node has already been decommissioned, rendering the Drop unnecessary.

Comment (xiaocai2333) [reply internal/datacoord/compaction_task_clustering.go:561]: DropCompactionPlan is idempotent on the datanode. The datanode merely keeps the task recorded in memory, so the operation simply attempts to remove the task from memory.

The DropCompactionPlan encapsulated in the session manager already handles the nodeNotFound error, so there is no need to handle it again when calling it externally.

**Comment by xiaocai2333** [comment]
rerun go-sdk

**Pull request merged**